### PR TITLE
Migration to recycler view

### PR DESCRIPTION
**Pull request by MikhailSkiyAtWork**
Fixed all conflicts

### CODE REVIEW

**Comment by oxied** [comment]
1. The same issue with images disappearing in lab57
2. Make launching of lab57 async.

File: app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.example.admin.lab_56_57.data.*;
+
+import java.util.List;
+
+/**
+ * Created by Admin on 22.05.2015.
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java]: Check your IDE user configuration

File: app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public int getItemCount(){
+        return(listOfItems_ != null ? listOfItems_.size() : 0);
+    }
+
+    /**
+     * Sets the image depends on the position
+     *
+     * @param holder
+     * @param imageId
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java]: Remove unnecessary javadoc fields.

File: app/src/main/java/com/example/admin/lab_56_57/data/CustomHolder.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.example.admin.lab_56_57.R;
+
+/**
+ * Created by Admin on 22.05.2015.
+ */
+public class CustomHolder extends RecyclerView.ViewHolder {
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/data/CustomHolder.java]: Name it properly.

File: app/src/main/java/com/example/admin/lab_56_57/data/CustomHolder.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    public TextView getTitle(){
+        return title_;
+    }
+
+    public TextView getDescription(){
+        return description_;
+    }
+
+    public ImageView getImage(){
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/data/CustomHolder.java]: Apply formatting to all of these three methods.

File: app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
+        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.big_list_item,null);
+        ViewHolder holder = new ViewHolder(view);
+        return holder;
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int i){
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java]: Apply formatting to this line.

File: app/src/main/java/com/example/admin/lab_56_57/data/CustomHolder.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package com.example.admin.lab_56_57.data;
```
Comment (oxied) [inline app/src/main/java/com/example/admin/lab_56_57/data/CustomHolder.java]: Should we delete this class in respect of ViewHolder?

File: app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java (hunk 5 above)
Comment (MikhailSkiyAtWork) [inline app/src/main/java/com/example/admin/lab_56_57/RecyclerAdapterForBigList.java]: Done

**Comment by oxied** [comment]
Launching of lab57 is still not async.

**Comment by oxied** [comment]
Done.

**Pull request merged**